From d07807ca68e7d3f0b94fdce9bc2a88a71d4324ef Mon Sep 17 00:00:00 2001
From: Owen W. Taylor <otaylor@fishsoup.net>
Date: Thu, 05 Apr 2012 17:10:23 +0000
Subject: gnome-settings-daemon: fix applying settings to newly added touchpads

In GTK+-3.4, touchpads no longer are classified as GDK_SOURCE_MOUSE,
they are GDK_SOURCE_TOUCHPAD, so we need to add a check for that
in the handling for newly added devices, or touchpads won't get
configured properly when returning from suspend.

Required GTK+ version is bumped to 3.3.18 for the GDK_SOURCE_TOUCHPAD
enum value.

https://bugzilla.gnome.org/show_bug.cgi?id=673595
--- gnome-settings-daemon-3.4.0.orig/configure.ac	2012-04-10 22:45:42.200069290 +0200
+++ gnome-settings-daemon-3.4.0/configure.ac	2012-04-10 22:45:42.340069297 +0200
@@ -44,7 +44,7 @@
 dnl ---------------------------------------------------------------------------
 
 GLIB_REQUIRED_VERSION=2.31.0
-GTK_REQUIRED_VERSION=3.3.4
+GTK_REQUIRED_VERSION=3.3.18
 GCONF_REQUIRED_VERSION=2.6.1
 GIO_REQUIRED_VERSION=2.26.0
 GNOME_DESKTOP_REQUIRED_VERSION=3.3.92
--- gnome-settings-daemon-3.4.0.orig/plugins/mouse/gsd-mouse-manager.c	2012-04-10 22:45:42.104069286 +0200
+++ gnome-settings-daemon-3.4.0/plugins/mouse/gsd-mouse-manager.c	2012-04-10 22:45:42.344069297 +0200
@@ -1019,7 +1019,10 @@
                  GdkDevice        *device,
                  GsdMouseManager  *manager)
 {
-        if (gdk_device_get_source (device) == GDK_SOURCE_MOUSE) {
+        GdkInputSource source;
+
+        source = gdk_device_get_source (device);
+        if (source == GDK_SOURCE_MOUSE || source == GDK_SOURCE_TOUCHPAD) {
                 if (run_custom_command (device, COMMAND_DEVICE_ADDED) == FALSE) {
                         set_mouse_settings (manager, device);
                 } else {
@@ -1039,7 +1042,10 @@
                    GdkDevice        *device,
                    GsdMouseManager  *manager)
 {
-        if (gdk_device_get_source (device) == GDK_SOURCE_MOUSE) {
+        GdkInputSource source;
+
+        source = gdk_device_get_source (device);
+        if (source == GDK_SOURCE_MOUSE || source == GDK_SOURCE_TOUCHPAD) {
                 int id;
 
                 run_custom_command (device, COMMAND_DEVICE_REMOVED);
